client/server/middleware/logger.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import morgan from 'morgan';
+
+/**
+ * Internal dependencies
+ */
+import { getLogger } from 'server/lib/logger';
+
+const NS_TO_MS = 1e-6;
+
+const logRequest = ( req, res, options ) => {
+	const { requestStart } = options;
+
+	// Equivalent to morgan's `dev` format
+	let fields = {
+		method: morgan.method( req ),
+		status: morgan.status( req, res ),
+		length: res.getHeader( 'Content-Length' ),
+		url: morgan.url( req ),
+		duration: Number(
+			( Number( process.hrtime.bigint() - requestStart ) * NS_TO_MS ).toFixed( 3 )
+		),
+	};
+
+	if ( process.env.NODE_ENV === 'production' ) {
+		// Standard Apache combined log output.
+		// Equivalent to morgan's `combined` format
+		fields = {
+			...fields,
+			remoteAddr: morgan[ 'remote-addr' ]( req ),
+			remoteUser: morgan[ 'remote-user' ]( req ),
```

Comment:
We can drop the `remote-user` field: it looks at the `Authorization` header and reports the username if `basic` HTTP auth is used. We never use that -- we do either OAuth or session cookies.

Reply:
That's great news. I started to re-implement morgan tokens manually as you suggest in another comment, but then found that the implementation of `remote-user` is not trivial. That's when I pivoted to reuse `morgan`'s tokens.

But if we can drop `remote-user`, I can go back to the original idea of reimplement the same tokens and get rid of `morgan`

Reply:
Now I realize that if we wanted, we could very easily find the real WP.com username in the cookie headers and log it. The `wordpress_logged_in` cookie has [format](https://github.com/WordPress/WordPress/blob/master/wp-includes/pluggable.php#L761):
```
username|expiration|token|hash
```
where the `username` token is simply the username, e.g., `jsnajdr`.

That works for the cookie auth that the web Calypso uses exclusively.